### DISCUSSION TITLE
force rpi4 to send over usb0?

### DISCUSSION THREAD

**Discussion by framlin** [iRobotEducation/create3_docs#274] [Questions] [answered]
I am going to insert a RPI4 into my brand new create3 and have both of them connected to my local WLAN, too. Because of this, I ask myself (and now you) a question:
How do I force the RPI4 to send ros2 messages over usb0 and how do I force the create3, to receive them from the usb-line?
At the same time I would like to read all ROS2 messages on my Ubuntu-PC, that is connected to the local WLAN, too. For example to display them with RVIZ2.
If - for example - the RPI4 sends on both channels and the create3 receives on both channels, I would think, that the create3 gets all messages twice and I think this would confuse the create3. Same would be true for the RPI4.

I assume, I am making a mistake in thinking, because I am totally new in ROS2. Please can you tell me, where I am wrong or what I have to do, to get this scenario working?

**Comment by justinIRBT** [comment, marked as answer]
This is discussed here:
https://iroboteducation.github.io/create3_docs/setup/xml-config/

You can see a section discussing using multiple network interfaces simultaneously for either Cyclone or FastDDS.  This is what you will be doing on the Raspberry Pi, using both wlan0 and usb0.  FastDDS will automatically use all the interfaces it finds on startup, with Cyclone (the default for galactic), you will have to supply a config file specifying both interfaces, that is covered in the link above.  Make sure your Create is configured to use the same RMW (Cyclone or Fast) that you are using on the pi, that is configured on the Create’s webserver.

**Reply by alsora** [reply]
I want to add that there is no risk of "getting all messages twice" when using multiple network interfaces.

**Reply by framlin** [reply]
At the moment, after I had attached the RPI4 via USB to the create3 and sending one time docking and undocking command from the RPI4, the creat3 is "out of order". The color-ring is solid red and even if I press the middle-button more than 7 secondonds, nothing changes.
I have unconnected the RPI4, but that does not help.

**Reply by justinIRBT** [reply]
Were you able to get past this?  If you are having trouble resetting the robot, the last resort is to take out the battery from the robot, leaving it unplugged for a couple of minutes, then plugging it back it and putting it on the dock to power on.  If the robot continues to go red from your order of operations, you can pull the log from the webserver and attach to help figure out what is going on.

**Reply by framlin** [reply]
It "healed" itself. After a long time (30 - 60 minutes) staying in the dock, it first switched to solid yellow and after another while it switched to solid white. 
After that I powered it off an on again and since then I did not had such problems any more.

**Comment by framlin** [comment]
Thank you very much for your answer!
